src/main/java/ch/naviqore/service/PublicTransportService.java

```diff
@@ -0,0 +1,31 @@
+package ch.naviqore.service;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Map;
+
+public interface PublicTransportService {
+    @NotNull List<StopTime> getNextDepartures(@NotNull Stop stop, @NotNull LocalDate from, @Nullable LocalDate until, int limit);
+
+    @NotNull List<Stop> getStops(@NotNull String like, @NotNull SearchType searchType);
+
+    @Nullable Stop getClosestStop(@NotNull Location location);
```

Comment:
Let's keep the wording the consistent: Nearest vs. closest - I would go for nearest 😄